include/nanobind/eigen/dense.h

```diff
@@ -12,9 +12,10 @@
 
 #include <nanobind/tensor.h>
 #include <Eigen/Core>
+#include <Eigen/SparseCore>
```

Comment:
Let's put the whole sparse-specific parts into a new headerfile `nanobind/eigen/sparse.h` that includes `nanobind/eigen/dense.h`

Reply:
This means that `is_eigen_sparse_v` and `is_eigen_xpr_v` have to be defined in a way that doesn't depend on elements in `Eigen/SparseCore` -- perhaps there is a constexpr field like `IsSparse` that can be used instead (or another field that is only defined for sparse types).

Reply:
I've moved everything to `nanobind/eigen/sparse.h`, but `is_eigen_sparse_v` cannot be moved (yet) unfortunately, because `is_eigen_v` matches `Eigen::MatrixBase` which `Eigen::SparseMatrix` inherits from. There is no member like `IsSparse` by we could maybe try to detect the `valuePtr()`, `outerIndexPtr()`, and `innerIndexPtr()` methods to decide. This would however entail SFINAE which is slow to compile

Reply:
`SparseMatrixBase` has a forward declaration in `Eigen/src/util/ForwardDeclarations.h`. Isn't it possible to do the test without pulling in the whole `SparseCore` header?

Reply:
Yes, this does work indeed. Good find! :ok_hand: